site/learn/expert-advice/identity-basics/slow-migration.md

```diff
@@ -0,0 +1,252 @@
+
+However, there are some issues or concerns with a big bang migration:
+
+* You will miss something during testing. After all, you aren’t doing this regularly; by definition this is a one time event. Plan for time to fix surprises.
+* Production systems, no matter how good the testing environment, have a way of being different, if for no other reason than that they are under load. You might encounter an unexpected error during the production rollout.
+* If something goes sideways, many users will be impacted, since they were all being migrated.
+* You'll write a bunch of code which you'll test intensely, use once and then throw away.
+* You need to ensure the new auth system "understands" how passwords were hashed in the old system. Or force everyone to reset their password, which is a subpar user experience, to put it mildly.
+
+Segment by segment migration, the second alternative, can be thought of as a series of "little bang" migrations. You split up your user data in a natural way, then you migrate each segment of users.
+
+A segment by segment migration lets you test your processes in production by migrating less critical, or more understanding, sets of users first. A set of employees or the engineering team may be a good first group, since they'll be more understanding of issues. You may be able to reuse some of your code or logic across segments. In general, this approach decreases risk when compared to the one time big bang migration.
+
+However, a segment by segment migration may be problematic in the following ways:
+
+* You may not be able to perform one if there's no natural divisions in your user base.
+* This may not be worth it if most of your users are in one segment. For example, if you have a popular application and a couple nascent apps and want to combine your users, you may not be decreasing your risk much.
+* This approach takes longer to complete, leaving you running multiple systems for a longer period of time.
+* You'll have downtime for each user segment, which requires additional off hours work or coordination.
+
+## How does a slow migration of auth data work?
+
+At a high level, a slow migration happens in four phases. Each user proceeds through the phases independently of any other.  Here's how the data flows before any changes are made: 
+
+{% plantuml source: _diagrams/learn/expert-advice/identity-basics/slow-migration/auth-before-migration.plantuml, alt: "The user auth process before migration." %}
+
+Then you'd stand up the new system, connect it to the old system, and route all authentication requests to the new system. When the new auth system is in front of the old auth system, the old system consulted the first time a user authenticates:
```

Comment:
"When the new auth system is in front of the old auth system, the old system consulted the first time a user authenticates" should be "When the new auth system is in front of the old auth system, the old system is consulted the first time a user authenticates".

I might also add some more information here that talks about how the new system is delegating or proxying to the old system, but only for that first login and how the new system then "copies" the data from the old system and stores it locally. This works to also keep the password intact since the new system has the plain-text password. 

Not sure if you cover this below, but talking about "upgrading" password hashing strength is something that fits here. If the old system was using MD5, the new system could use BCrypt.